### PR TITLE
fetch-metadata-demo

### PR DESCRIPTION
**Pull request by krisbitney**
This PR adds a demo demonstrating how to fetch wrapper metadata with the client.

Screen Shot 2022-05-16 at 3 43 26 PM

Closes https://github.com/polywrap/demos/issues/32

Related: https://github.com/polywrap/documentation/issues/105

### CODE REVIEW

**Review by cbrzn — APPROVED** [review]
Looks awesome! I have two thoughts:
- Should we add in the README a Wrap URI (for example, the Uniswap V3) so the user doesn't need to look for one?
- I noticed that currently if a URI is valid but doesn't have the meta manifest, we show `Failed to resolve URI` message. Should we instead explicitly say that the URI was valid but it doesn't have this `meta` manifest? :thinking:

File: fetch-metadata/src/util/MetaData.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {MetaManifest} from "@web3api/client-js";
+
+export interface MetaData extends MetaManifest {
+  iconImage?: string;
+  links?: {
```
Comment (dOrgJelli) [inline fetch-metadata/src/util/MetaData.ts:5]: Why are we doing this? Shouldn't we just use the `MetaManifest` type directly, without these added fields?

**Review by dOrgJelli — CHANGES_REQUESTED** [review]
This looks amazing! I agree we should add some example URI's the user can simply copy/paste into the search box for testing. Could just add these below the search box as some examples.

Also the page layout doesn't quite work for me:
image

File: fetch-metadata/src/util/MetaData.ts (hunk 1 above)
Comment (krisbitney) [reply fetch-metadata/src/util/MetaData.ts:5]: The `icon` and `path` properties in the meta manifest have paths from the wrapper root to the image files. The iconImage properties contain base64 string representations of the image data, which is fetched from the paths. Organizing it in the same object made it easier to map over the links array and display the images.

File: fetch-metadata/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "react-scripts": "3.4.4",
+    "typescript": "~4.6.4"
+  },
+  "devDependencies": {
+    "@types/node": "~16.11.35",
+    "@types/react": "~17.0.45",
+    "@types/react-dom": "~17.0.17",
+    "@web3api/cli": "latest"
```
[the remark this answers is not in the document]
Comment (krisbitney) [reply fetch-metadata/package.json]: fixed :)

**Comment by krisbitney** [comment]
> Looks awesome! I have two thoughts:
> 
> * Should we add in the README a Wrap URI (for example, the Uniswap V3) so the user doesn't need to look for one?
> * I noticed that currently if a URI is valid but doesn't have the meta manifest, we show `Failed to resolve URI` message. Should we instead explicitly say that the URI was valid but it doesn't have this `meta` manifest? 🤔

I updated the readme and improved the error message

File: fetch-metadata/src/util/MetaData.ts (hunk 1 above)
Comment (krisbitney) [reply fetch-metadata/src/util/MetaData.ts]: I refactored to use the `MetaManifest` type directly

**Comment by krisbitney** [comment]
> This looks amazing! I agree we should add some example URI's the user can simply copy/paste into the search box for testing. Could just add these below the search box as some examples.
> 
> Also the page layout doesn't quite work for me: image

I think I've fixed the layout issue but I'm not certain because I don't have a higher resolution screen to test with.

**Review by dOrgJelli — APPROVED** [review]
Looks amazing @krisbitney 👏

**Pull request merged**